inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/InweboAuthenticator.java

```diff
@@ -0,0 +1,240 @@
+                    int retryCount = Integer.parseInt(authenticatorProperties.get(InweboConstants.RETRYCOUNT));
+                    int retryInterval = Integer.parseInt(authenticatorProperties.get(InweboConstants.RETRINTERVAL));
+
+                    if (userId != null && serviceId != null && p12file != null && p12password != null) {
+                        PushREST push;
+                        push = new PushREST(serviceId, p12file, p12password, userId, retryCount, retryInterval);
+                        pushResponse = push.run();
+                        PushREST.showServlet(response, relyingParty, type, finalReferer);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            throw new AuthenticationFailedException(e.getMessage(), e);
+        }
+
+
+        return;
+    }
+
+    /**
+     * Get the configuration properties of UI
+     */
+    @Override
+    public List<Property> getConfigurationProperties() {
+        List<Property> configProperties = new ArrayList<Property>();
+        Property userId = new Property();
+
+        userId.setName(InweboConstants.USER_ID);
+        userId.setDisplayName("User name");
+        userId.setRequired(true);
+        userId.setDescription("Enter your Inwebo UserName");
+        configProperties.add(userId);
+
+        Property serviceId = new Property();
+        serviceId.setName(InweboConstants.SERVICE_ID);
+        serviceId.setDisplayName("Service Id");
+        serviceId.setRequired(true);
+        serviceId.setDescription("Enter your service Id");
+        configProperties.add(serviceId);
+
+        Property p12file = new Property();
+        p12file.setName(InweboConstants.INWEBO_P12FILE);
+        p12file.setDisplayName("Certificate File");
+        p12file.setRequired(true);
+        p12file.setDescription("Enter your p12_file path");
+        configProperties.add(p12file);
+
+        Property p12password = new Property();
+        p12password.setName(InweboConstants.INWEBO_P12PASSWORD);
+        p12password.setDisplayName("Certificate Password");
+        p12password.setRequired(true);
+        p12password.setConfidential(true);
+        p12password.setDescription("Enter your p12_password");
+        configProperties.add(p12password);
+
+        Property retryCount = new Property();
+        retryCount.setName(InweboConstants.RETRYCOUNT);
+        retryCount.setDisplayName("Retry Count");
+        retryCount.setRequired(true);
+        retryCount.setDescription("Number of attempts waiting for authentication(<10)");
+        configProperties.add(retryCount);
+
+        Property retryInterval = new Property();
+        retryInterval.setName(InweboConstants.RETRINTERVAL);
+        retryInterval.setDisplayName("Retry Interval");
+        retryInterval.setRequired(true);
+        retryInterval.setDescription("Period of time for waiting eg:1000");
+        configProperties.add(retryInterval);
+        return configProperties;
```

Comment:
Adhere to a convention on the Displayname and description. (either make first letter in each word capital in all or not.) UI needs to be consistent.